services/property-svc/internal/property/projections/property_postgres_projection/property_postgres_projection.go

```diff
@@ -0,0 +1,614 @@
+	p := &Projection{custom.NewCustomProjection(es, subscriptionGroupName), property_repo.New(hwdb.GetDB())}
+	p.initEventListeners()
+	return p
+}
+
+func (p *Projection) initEventListeners() {
+	p.RegisterEventListener(propertyEventsV1.PropertyCreated, p.onPropertyCreated)
+	p.RegisterEventListener(propertyEventsV1.PropertyDescriptionUpdated, p.onPropertyDescriptionUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertySetIDUpdated, p.onPropertySetIDUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertySubjectTypeUpdated, p.onSubjectTypeUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeUpdated, p.onFieldTypeUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyNameUpdated, p.onNameUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyArchived, p.onPropertyArchived)
+	p.RegisterEventListener(propertyEventsV1.PropertyRetrieved, p.onPropertyRetrieved)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataCreated, p.onPropertyFieldTypeDataCreated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataAllowFreetextUpdated, p.onAllowFreetextUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataSelectOptionsRemoved, p.onFieldTypeDataSelectOptionsRemoved)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataSelectOptionsUpserted, p.onFieldTypeDataSelectOptionsUpserted)
+}
+
+func (p *Projection) onPropertyCreated(ctx context.Context, evt hwes.Event) (error, esdb.NackAction) {
+	log := zlog.Ctx(ctx)
+	db := hwdb.GetDB()
+
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	defer func() {
+		err := tx.Rollback(ctx)
+		if !errors.Is(err, pgx.ErrTxClosed) {
+			log.Error().Err(err).Msg("rollback failed.")
+		}
+	}()
+
+	propertyRepo := property_repo.New(db).WithTx(tx)
+
+	var payload propertyEventsV1.PropertyCreatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
+		return err, esdb.NackActionRetry
+	}
+
+	// Create FieldTypeData
+	fieldTypeDataID, err := propertyRepo.CreateFieldTypeData(ctx)
+	if err := hwdb.Error(ctx, err); err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	// Create Property
+	propertyID, err := uuid.Parse(payload.ID)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	value, found := pb.SubjectType_value[payload.SubjectType]
+	if !found {
+		return fmt.Errorf("subject_type %s invalid", payload.SubjectType), esdb.NackActionRetry
+	}
+	subjectType := (pb.SubjectType)(value)
+
+	value, found = pb.FieldType_value[payload.FieldType]
+	if !found {
+		return fmt.Errorf("field_type %s invalid", payload.FieldType), esdb.NackActionRetry
+	}
+	fieldType := (pb.FieldType)(value)
+
+	err = propertyRepo.CreateProperty(ctx, property_repo.CreatePropertyParams{
+		ID:              propertyID,
+		SubjectType:     int32(subjectType),
+		FieldType:       int32(fieldType),
+		Name:            payload.Name,
+		Description:     "",
```

Comment:
do we not get the description here? we should, right? (change proto if needed)

Reply:
If we create a property with a description, which we can, we are dispatching the `PropertyCreated` event as well as the `PropertyDescriptionUpdatedEvent` which then updates the description of the property.
This is exactly how we do it in the task projection as well. I will remove the description as well as the isArchived from the sql statement.